src/expire.c

```diff
@@ -489,6 +489,47 @@ int checkAlreadyExpired(long long when) {
     return (when <= mstime() && !server.loading && !server.masterhost);
 }
 
+#define FLAG_NO_FLAGS 0
+#define FLAG_NX (1<<0)
+#define FLAG_XX (1<<1)
+#define FLAG_GT (1<<2)
+#define FLAG_LT (1<<3)
+
+/* Parse additional flags of expire commands
+ *
+ * Supported flags:
+ * - NX: set expiry only when the key has no expiry
+ * - XX: set expiry only when the key has an existed expiry
+ * - GT: set expiry only when the new expiry is greater than current one
+ * - LT: set expiry only when the new expiry is less than current one */
+void parseExtendedExpireArguments(client *c, int *flags) {
+    // command index
+    int j = 3;
+
+    for (; j < c->argc; j++) {
```

Comment:
we don't use line comments `//` in redis.
was gonna change the comment style, but i think it isn't really needed, so eliminated a few lines.
```suggestion
    for (int j = 3; j < c->argc; j++) {
```